cmake/FindCSharp.cmake

```diff
@@ -0,0 +1,87 @@
+##
+## $Id$
+## It is part of the SolidOpt Copyright Policy (see Copyright.txt)
+## For further details see the nearest License.txt
+##
+
+## This file is based on the work of SimpleITK:
+##  https://github.com/SimpleITK/SimpleITK/blob/master/CMake/FindCSharp.cmake
```

Comment:
We can't copy and paste code from other projects into Detours, this code has it's own LICENSE.

Reply:
For this, I'll see if we can solve the problem by referencing the third party staffs as submodule or some other ways...

Reply:
CMake 3.8+ has native support for C#
See: https://cmake.org/cmake/help/latest/release/3.8.html#id4